RPIPs/RPIP-44.md

```diff
@@ -12,22 +12,45 @@ requires: eip-7002
+  - This function MAY be called by any user if the Node has a `deficit` greater than `exit_deficit`
+    - It MUST NOT be possible to exit more than the minimum number of validators needed to cover `deficit`
+    - The caller SHALL receive `keeper_reward`, which is deducted from the Node Operator's share
+- `deficit` is `balance_deficit + penalties - expected_exit_balance`
+  - `balance_deficit = 32-balance`; this represents the amount of validator principal lost due to Ethereum penalties
+  - `penalties`; this represents the amount of ETH owed to the protocol per [RPIP-42's penalties](RPIP-42.md#penalizable-offenses)
+  - `expected_credit_from_exit`; this represents the amount of ETH the NO expects to receive as their share from validator exits that have begun but not yet completed
+    - The protocol SHALL update `expected_credit_from_exit` when exits are begun, and subtract the minimum 1 ETH if the validator was slashed
+    - The protocol SHALL update `expected_credit_from_exit` when exits are completed 
+- The protocol SHOULD use `rewards` and/or `credit` held by the NO to decrease `debt` by the same amount prior to taking action on exits
+- The protocol SHOULD use staked RPL held by the NO to decrease `debt` by the corresponding amount prior to taking action on exits
+- The initial settings SHALL be:
+  - `exit_deficit`: 0.5 ETH
+  - `keeper_reward`: 0.01 ETH + approximate gas refund based on precalculated gas and the BASEFEE opcode
+
+## Rationale
+- `exit_deficit` is about enough for over 3 months covering leakage at 4% apy, using the rule of thumb that losses accrue at half the speed of gains `32*(.04/2)*(3/12)` = .16 ETH. Users that have accrued this much offline time are likely to be unable to recover their validators at all due to incapacitation, lost keys, etc.
+- `keeper_reward` is set to ensure that users are incentivized to call this function when possible
+- `expected_credit_from_exit` has a known potential shortfall in cases of correlated slashing; that said, upon the completion of the exits, the numbers will be correctly updated. This means there may be a second round of exits in such a case.
+
+## Risk Analysis
+
+### Forced Exits During Inactivity Leaking
+
+The Ethereum staking formula includes additional penalties for offline validators when large portions of the network are offline. This state of accelerated punishment is called ["inactivity leaking"](https://ethereum.org/en/developers/docs/consensus-mechanisms/pos/rewards-and-penalties/#inactivity-leak). With the RP protocol forcing exits for validators under a 31.9 ETH balance, validators who are offline during an inactivity leaking scenario may be exited even if the validator is only offline for a brief period. In some extreme scenarios, a small number of missed attestations could cause a validator to be ejected from the network. This may be necessary, however, in order to protect the value of rETH during an unusual network state for Ethereum.
```

Comment:
31.9 ETH does not align with the `exit_deficit` value of 0.5 ETH.

Reply:
badly written in general -- merged a per-validator and a per-node thing here